packages/contractkit/src/identity/metadata.ts

```diff
@@ -38,11 +39,17 @@ export class IdentityMetadataWrapper {
   }
 
   static async fetchFromURL(kit: ContractKit, url: string) {
-    const resp = await fetch(url)
-    if (!resp.ok) {
-      throw new Error(`Request failed with status ${resp.status}`)
-    }
-    return this.fromRawString(kit, await resp.text())
+    return retryAsyncWithBackOff(
+      async () => {
+        const resp = await fetch(url)
+        if (!resp.ok) {
```

Comment:
Thoughts on a circuit breaker for 404? `resp.ok` will be false on a 404, but we don't want to retry.

Reply:
great point!

Reply:
Looking at the fetch docs I see any `4XX` code will make `resp.ok` false. Worth noting